modules/sdk-core/src/account-lib/mpc/tss/ecdsa/types.ts

```diff
@@ -37,13 +69,19 @@ export interface XShare {
   y: string; // combined public key
   x: string; // combined secret
   chaincode: string;
+  ntilde?: string;
```

Comment:
Similar to my comments about the `BaseCurve` vs. `BaseCurveWithOrder` type, I think these groups of optional fields should be distinguished by a more specific type (in this case something like `export type XShareWithNTilde = XShare & NTilde;` to eliminate use of `!`s on fields elsewhere in the code. Then we can add a type check function that is something like `function hasNTilde(xShare: XShare): xShare is XShareWithNTilde  { return 'ntilde' in xShare; }` to guard code that requires the additional fields.

Reply:
Done